[CI] CoordinatorTests.testLogsWarningPeriodicallyIfClusterNotFormed failing on 7.x

Logs: https://gradle-enterprise.elastic.co/s/n2zxbakfbzi36/console-log?task=:server:test

Couldn't reproduce locally with:
```
./gradlew ':server:test' --tests "org.elasticsearch.cluster.coordination.CoordinatorTests.testLogsWarningPeriodicallyIfClusterNotFormed" \
  -Dtests.seed=4223C526497B2D65 \
  -Dtests.security.manager=true \
  -Dtests.locale=no \
  -Dtests.timezone=Asia/Chungking \
  -Dcompiler.java=12 \
  -Druntime.java=8
```

Stacktrace:
```

org.elasticsearch.cluster.coordination.CoordinatorTests > testLogsWarningPeriodicallyIfClusterNotFormed FAILED
--
java.lang.AssertionError: [{node0}{zL-LbQAAQACH3l2B_____w}{0.0.0.0}{0.0.0.0:20}{dim}, {node2}{77fEz___T_-jSIsAAAAAAA}{0.0.0.0}{0.0.0.0:22}{dim}] vs [{node0}{zL-LbQAAQACH3l2B_____w}{0.0.0.0}{0.0.0.0:20}{dim}, {node1}{J6nlEgAAQACtOVyt_____w}{0.0.0.0}{0.0.0.0:16}{dim}, {node2}{77fEz___T_-jSIsAAAAAAA}{0.0.0.0}{0.0.0.0:22}{dim}]
Expected: <[{node0}{zL-LbQAAQACH3l2B_____w}{0.0.0.0}{0.0.0.0:20}{dim}, {node1}{J6nlEgAAQACtOVyt_____w}{0.0.0.0}{0.0.0.0:16}{dim}, {node2}{77fEz___T_-jSIsAAAAAAA}{0.0.0.0}{0.0.0.0:22}{dim}]>
but: was <[{node0}{zL-LbQAAQACH3l2B_____w}{0.0.0.0}{0.0.0.0:20}{dim}, {node2}{77fEz___T_-jSIsAAAAAAA}{0.0.0.0}{0.0.0.0:22}{dim}]>
at __randomizedtesting.SeedInfo.seed([4223C526497B2D65:7FE5C0D97B91B710]:0)
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
at org.junit.Assert.assertThat(Assert.java:956)
at org.elasticsearch.cluster.coordination.CoordinatorTests$3.assertMatched(CoordinatorTests.java:1250)
at org.elasticsearch.test.MockLogAppender.assertAllExpectationsMatched(MockLogAppender.java:66)
at org.elasticsearch.cluster.coordination.CoordinatorTests.testLogsWarningPeriodicallyIfClusterNotFormed(CoordinatorTests.java:1255)

```

--- a/server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java
@@ -1215,8 +1215,14 @@ public class CoordinatorTests extends AbstractCoordinatorTestCase {
                 clusterNode.disconnect();
             }
 
-            cluster.runFor(defaultMillis(LEADER_CHECK_INTERVAL_SETTING) + defaultMillis(LEADER_CHECK_TIMEOUT_SETTING),
+            cluster.runFor(defaultMillis(LEADER_CHECK_TIMEOUT_SETTING) // to wait for any in-flight check to time out
+                    + defaultMillis(LEADER_CHECK_INTERVAL_SETTING) // to wait for the next check to be sent
+                    + 2 * DEFAULT_DELAY_VARIABILITY, // to send the failing check and receive the disconnection response
                 "waiting for leader failure");
+
+            for (final ClusterNode clusterNode : cluster.clusterNodes) {
+                assertThat(clusterNode.getId() + " is CANDIDATE", clusterNode.coordinator.getMode(), is(CANDIDATE));
+            }
 
             for (int i = scaledRandomIntBetween(1, 10); i >= 0; i--) {
                 final MockLogAppender mockLogAppender = new MockLogAppender();

